File: antd-cpp/examples/06-private-data.cpp
#include <iostream>
#include <string>
#include <vector>
#include "antd/antd.hpp"

int main() {
    try {
        antd::Client client;

        // Store private (encrypted) data
        std::string secret = "Sensitive information";
        std::vector<uint8_t> data(secret.begin(), secret.end());

        auto result = client.data_put_private(data);
        std::cout << "Private data stored\n";
        std::cout << "Data map: " << result.address << "\n";
        std::cout << "Cost: " << result.cost << " atto\n";

        // Retrieve using the data map
        auto retrieved = client.data_get_private(result.address);
        std::string text(retrieved.begin(), retrieved.end());
        std::cout << "Retrieved: " << text << "\n";

    } catch (const antd::NotFoundError& e) {
        std::cerr << "Not found: " << e.what() << "\n";
        return 1;
    } catch (const antd::PaymentError& e) {
        std::cerr << "Payment failed: " << e.what() << "\n";
        return 1;
    } catch (const antd::AntdError& e) {
        std::cerr << "Error: " << e.what() << "\n";
        return 1;
    }
    return 0;
}

File: antd-ruby/examples/06_private_data.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Example 06: Private data put/get

require_relative "../lib/antd"

client = Antd::Client.new

# Store private (encrypted) data
result = client.data_put_private("sensitive information")
puts "Private data stored (cost: #{result.cost} atto)"
puts "Data map: #{result.address}"

# Retrieve private data using the data map
data = client.data_get_private(result.address)
puts "Retrieved: #{data}"
